src/input/keyboard_and_mouse.rs

```diff
@@ -77,32 +99,36 @@ impl Input for KeyboardAndMouse {
                 mouse::Event::CursorReturned => {
                     self.is_cursor_taken = false;
                 }
-                mouse::Event::Input {
-                    button: mouse::Button::Left,
-                    state,
-                } => match state {
-                    ButtonState::Pressed => {
-                        self.is_mouse_pressed = !self.is_cursor_taken;
-                    }
-                    ButtonState::Released => {
-                        if !self.is_cursor_taken && self.is_mouse_pressed {
-                            self.left_clicks.push(self.cursor_position);
+                mouse::Event::Input { state, button } => {
+                    match state {
+                        ButtonState::Pressed => {
+                            if self.is_cursor_taken {
```

Comment:
This should be `!self.is_cursor_taken`. We want to avoid capturing button presses and clicks when the cursor is currently taken. Maybe the name `is_cursor_taken` is a bit confusing? We could think about changing the name of the events too. Not a fan of `CursorReturned` and `CursorTaken`, personally.